app/src/org/commcare/models/database/HybridFileBackedSqlStorage.java

```diff
@@ -292,11 +296,15 @@ public void write(Persistable persistable) {
             if (dataFilePath != null) {
                 HybridFileBackedSqlHelpers.unsetFileAsOrphan(db, dataFilePath);
             }
-            db.setTransactionSuccessful();
+            if(startedTransaction) {
```

Comment:
We'll actually be in a transaction either way, quite likely, since there is often a transaction wrapping this method call. This would be evaluating the status of a secondary transaction.

The reason we need to use a transaction in this method independent of the outer transaction is that there's a specific secondary (non-database call) side effect being tracked from the Orphan handlers.